### ISSUE TITLE
Porter archive fails to archive from a 127.0.0.0/8 insecure registry

### ISSUE THREAD

**Issue by bherw** [getporter/porter#1813]
## Describe the bug
Archiving from insecure registries on loopback IPs other than 127.0.0.1 seems to fail despite specifying the --insecure-registry flag.

Note the docker insecure registries config includes this range by default, so docker should not be the one throwing this error:
```shell
docker info --format 'Insecure Registries: {{.RegistryConfig.InsecureRegistryCIDRs}}'
Insecure Registries: [127.0.0.0/8]
```

## To Reproduce
1. Run an insecure registry on the loopback adaptor on a random IP: `docker run -p 127.5.34.33:39693:5000 -d registry`
2. Use some porter app with the registry
3. Run `porter publish --insecure-registry` to publish to that registry
4. Try to archive that newly published bundle, e.g. `porter archive porter-app.tgz --insecure-registry --reference 127.5.34.33:39693/test-echo:v0.1.0`
5. See error

## Expected behavior
The archive command should succeed.

## Porter Command and Output
```
$ porter archive porter-app.tgz --insecure-registry --reference 127.5.34.33:39693/test-echo:v0.1.0
Error: error preparing artifacts: Get "https://127.5.34.33:39693/v2/": http: server gave HTTP response to HTTPS client
```

## Version
porter v0.38.7 (43a2b2ae)

**Comment by Jemgoss** [comment]
I've hit this same issue using different steps but I believe it's related to the same issue as this and *is not fixed by* https://github.com/getporter/porter/pull/2273. I've tried a build of porter on that branch and it still fails.

My steps to reproduce:

1. Alias a hostname other than "localhost" to the loopback address. I have this in my /etc/hosts file:
```
127.0.0.1       localhost bundleregistry
```

2. Start a local registry binding to default 0.0.0.0 (I'm using port 5001 but I don't think that's relevant).
```
$ docker run -d -p 5001:5000 --name bundle-registry registry:2
$ curl http://bundleregistry:5001/v2/_catalog
{"repositories":[]}
```

3.  Now try to publish a TGZ to the aliased registry:
```
$ porter publish -a bun.tgz -r bundleregistry:5001/test:v0.0.1 --insecure-registry
Beginning bundle publish to bundleregistry:5001/test:v0.0.1. This may take some time.
unable to push updated image: unable to push image bundleregistry:5001/test:3a3921a3fca03e4dca685528a927449c: failed to write image sha256:92679cb2412b63d073e9e2b9c3959c5253a3f0d80294faeab726487bd6b4ce32 to bundleregistry:5001/test:3a3921a3fca03e4dca685528a927449c: failed to write image bundleregistry:5001/test:3a3921a3fca03e4dca685528a927449c: Get "https://bundleregistry:5001/v2/": http: server gave HTTP response to HTTPS client
```

4.  But using "localhost" works:
```
$ porter publish -a bun.tgz -r localhost:5001/test:v0.0.1 --insecure-registry
Beginning bundle publish to localhost:5001/test:v0.0.1. This may take some time.
Starting to copy image localhost:5001/test:68516b03a700e3beb53eff53400141d5@sha256:92679cb2412b63d073e9e2b9c3959c5253a3f0d80294faeab726487bd6b4ce32...
Completed image localhost:5001/test:68516b03a700e3beb53eff53400141d5@sha256:92679cb2412b63d073e9e2b9c3959c5253a3f0d80294faeab726487bd6b4ce32 copy
Starting to copy image localhost:5001/test:16bfd780b6d9d6b1d67be18f3c207bb2@sha256:d2b5fedbbf0c15f78f41db7afb34b2af4215f15debc9438ddc861d0361c874e2...
Completed image localhost:5001/test:16bfd780b6d9d6b1d67be18f3c207bb2@sha256:d2b5fedbbf0c15f78f41db7afb34b2af4215f15debc9438ddc861d0361c874e2 copy
Bundle tag localhost:5001/test:v0.0.1 pushed successfully, with digest "sha256:0eaa28ed045037a70a133c48a72538ebe94c8b67191a2b8f09178f67387a4289"
```

After some investigation, I've traced this to yet more inconsistent behavior in another dependent library `github.com/google/go-containerregistry`.

At line 78 in `https://github.com/google/go-containerregistry/blob/main/pkg/v1/remote/transport/ping.go` the reg.Scheme() is "https" despite the --insecure-registry flag in porter, so it appears the property isn't getting passed through to this layer.

Comments in this ping() function and NewWithContext() in transport.go suggest this package uses its own "localhost heuristic" to determine whether to use https/TLS or plain text. I have yet to trace the call path all the way through to this function from porter, but I know that when I hack the ping() function to add an "http" scheme, things work correctly.

**Comment by carolynvs** [comment]
Sadly every single dependency that we use to interact with external systems has their own way (or not) of allowing us to configure insecure registries. I have created a new issue to track what you are running into since this original issue was about non-standard loopback addresses and yours is more about how to handle/configure an insecure registry with a custom name.

**Comment by Jemgoss** [comment]
Ok, I had assumed both issues were part of the docker.MatchLocalhost() test.

**Comment by carolynvs** [comment]
We'll see! 🤷‍♀️

**Comment by carolynvs** [comment]
We've got part of the fix in for insecure registries, don't worry I'm in the middle of getting a fix for #2297 too.
